mlonmcu/target/riscv/ara.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (c) 2022 TUM Department of Electrical and Computer Engineering.
+#
+# This file is part of MLonMCU.
+# See https://github.com/tum-ei-eda/mlonmcu.git for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""MLonMCU ARA Target definitions"""
+
+import os
+import re
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from mlonmcu.logging import get_logger
+from mlonmcu.feature.features import SUPPORTED_TVM_BACKENDS
+from mlonmcu.target.common import cli, execute
+from mlonmcu.target.metrics import Metrics
+from .riscv import RISCVTarget
+from .util import update_extensions
+
+logger = get_logger()
+
+
+class AraTarget(RISCVTarget):
+    """Target using a Pulpino-like VP running in the GVSOC simulator"""
+
+    FEATURES = RISCVTarget.FEATURES + ["log_instrs"]
+
+    DEFAULTS = {
+        **RISCVTarget.DEFAULTS,
+        "xlen": 64,
+        "abi": "lp64d",
+        "extensions": ["g", "c", "v"],  # TODO overwrite extensions elegantly
+        "nr_lanes": 4,
+        "vlen": 4096,
```

Comment:
This will be overwritten by feature vext

Reply:
To make use of the Vext feature it would make sense to disable the “v” extension by default in this target. However I would imagine that something in the assembly startup script might break when compiling with Vext-support because it probably touches Vext-specific CSRs (will have to double-check)

Reply:
If there are such problems, let’s just do not allow running the `ara` target without the `vext` feature being enabled e.g. `assert self.enable_vext, "ARA target needs vext feature"`

Reply:
For consistency with the other riscv targets, it would also make sense not not overwrite `mabi` and instead enforce (i.e. using assertions) `fpu: double` and `xlen: 64`

Reply:
yes...

Reply:
But actually, the compilation without "v" extension seems to be fine. However, vlen is needed for the hardware compilation.